ethereum/core/src/test/java/org/hyperledger/besu/ethereum/vm/operations/JumpOperationTest.java

```diff
@@ -136,17 +143,48 @@ public void shouldHaltWithInvalidJumDestinationWhenLocationIsOutsideOfCodeRange(
   @Test
   public void longContractsValidate() {
     final JumpOperation operation = new JumpOperation(gasCalculator);
+    final Bytes longCode =
+        Bytes.fromHexString(
+            "0x60006000351461001157600050610018565b6101016020525b60016000351461002a5760005061002f565b326020525b60026000351461004157600050610046565b336020525b6003600035146100585760005061005d565b306020525b60046000351461006f57600050610075565b60016020525b60005160005260006020351461008d576000506100b6565b5a600052602051315060165a60005103036000555a600052602051315060165a60005103036001555b6001602035146100c8576000506100f1565b5a6000526020513b5060165a60005103036000555a6000526020513b5060165a60005103036001555b6002602035146101035760005061012c565b5a6000526020513f5060165a60005103036000555a6000526020513f5060165a60005103036001555b60036020351461013e5760005061017a565b6106a5610100525a600052602060006101006020513c60205a60005103036000555a600052602060006101006020513c60205a60005103036001555b00");
+
     final MessageFrame longContract =
         createMessageFrameBuilder(Gas.of(100))
             .pushStackItem(UInt256.fromHexString("0x12c"))
-            .code(
-                new Code(
-                    Bytes.fromHexString(
-                        "0x60006000351461001157600050610018565b6101016020525b60016000351461002a5760005061002f565b326020525b60026000351461004157600050610046565b336020525b6003600035146100585760005061005d565b306020525b60046000351461006f57600050610075565b60016020525b60005160005260006020351461008d576000506100b6565b5a600052602051315060165a60005103036000555a600052602051315060165a60005103036001555b6001602035146100c8576000506100f1565b5a6000526020513b5060165a60005103036000555a6000526020513b5060165a60005103036001555b6002602035146101035760005061012c565b5a6000526020513f5060165a60005103036000555a6000526020513f5060165a60005103036001555b60036020351461013e5760005061017a565b6106a5610100525a600052602060006101006020513c60205a60005103036000555a600052602060006101006020513c60205a60005103036001555b00")))
+            .code(new Code(longCode, Hash.hash(longCode)))
             .build();
     longContract.setPC(255);
 
     final OperationResult result = operation.execute(longContract, null);
     assertThat(result.getHaltReason()).isEmpty();
   }
+
+  @Test
+  public void shouldReuseJumpDestMap() {
+    final JumpOperation operation = new JumpOperation(gasCalculator);
+    Bytes jumpBytes = Bytes.fromHexString("0x6003565b00");
+    Code toRun = spy(new Code(jumpBytes, Hash.hash(jumpBytes)));
+    MessageFrame frame =
+        createMessageFrameBuilder(Gas.of(10_000))
+            .pushStackItem(UInt256.fromHexString("0x03"))
+            .code(toRun)
+            .build();
+    frame.setPC(CURRENT_PC);
+
+    OperationResult result = operation.execute(frame, evm);
+    assertThat(result.getHaltReason()).isEmpty();
+    Mockito.verify(toRun, times(1)).calculateJumpDests();
+
+    // do it again to prove we don't recalc
```

Comment:
Optional: A cache-miss (new code, maybe "0x02") and a `Mockito.verify(toRun, times(2)).calculateJumpDests();` would make it clear on reading that the verify doesn't reset between checks.

Reply:
That would not pass though, since we are spying on the Code instance. A new code being analyzed would not cause toRun to be re-analyzed. If we want to test explicitly for cache hits/misses, I can add that, but this is really watching to see that for a given Code which contains the Jump operation, the analysis is only run once.